File: mo-system/mo-system-api/src/main/java/ink/ckx/mo/system/api/model/vo/role/RoleDetailVO.java
package ink.ckx.mo.system.api.model.vo.role;

import ink.ckx.mo.common.web.enums.StatusEnum;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/15
 */
@Schema(description = "角色详情VO")
@Data
public class RoleDetailVO {

    @Schema(description = "角色ID")
    private Long id;

    @Schema(description = "角色名称")
    private String name;

    @Schema(description = "角色编码")
    private String code;

    @Schema(description = "排序")
    private Integer sort;

    @Schema(description = "数据权限")
    private Integer dataScope;

    @Schema(description = "状态")
    private StatusEnum status;
}

File: mo-admin/mo-admin-api/src/main/kotlin/ink/ckx/mo/admin/api/model/vo/role/RoleDetailVO.kt
package ink.ckx.mo.admin.api.model.vo.role

import ink.ckx.mo.common.web.enums.StatusEnum
import io.swagger.v3.oas.annotations.media.Schema

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/15
 */
@Schema(description = "角色详情VO")
data class RoleDetailVO(

    @Schema(description = "角色ID")
    var id: Long? = null,

    @Schema(description = "角色名称")
    var name: String? = null,

    @Schema(description = "角色编码")
    var code: String? = null,

    @Schema(description = "排序")
    var sort: Int? = null,

    @Schema(description = "数据权限")
    var dataScope: Int? = null,

    @Schema(description = "状态")
    var status: StatusEnum? = null,
)